hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OmLockOperation.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.lock;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+import org.apache.hadoop.util.Time;
+import org.apache.ratis.util.function.CheckedBiFunction;
+import org.apache.ratis.util.function.CheckedFunction;
+
+/**
+ * Manage locking of volume, bucket, keys and others.
+ */
+public class OmLockOperation {
+  private static final long LOCK_TIMEOUT_DEFAULT = 10 * 60 * 1000;
```

Comment:
Why it needs a 10min timeout?

Reply:
Idea here is,
- Hadoop RPC thread pool  `ozone.om.grpc.read.thread.num` is configured default 10, so it may be blocked if do not timeout.
- number of request waiting keeps increasing and hence memory

So IMO, a server should timeout request at appropriate time. May be lesser value as configurable or context of usages.

Reply:
OM currently does not have lock timeout.  We do not see any problems.  Let's don't fix this non-occurring problem here for simplicity reason.  If we see the problem, we can add lock timeout.  @kerneltime , what do you think?

Initially, I thought lock time would complicate the code a lot.  Now, I find out that It is actually not very complicated.  Thanks @errose28 for pointing it out.